src/main/common/kalman.c

```diff
@@ -139,21 +107,12 @@ FAST_CODE float kalman_process(kalman_t* kalmanState, float input, float target)
 }
 
 
-void FAST_CODE kalman_update(float* input, float* output)
+float FAST_CODE kalman_update(float input, int axis)
 {
 
-    setPoint[X] = getSetpointRate(X);
-    setPoint[Y] = getSetpointRate(Y);
-    setPoint[Z] = getSetpointRate(Z);
-
-    update_kalman_covariance(input);
-    output[X] = kalman_process(&kalmanFilterStateRate[X], input[X], setPoint[X] );
-    output[Y] = kalman_process(&kalmanFilterStateRate[Y], input[Y], setPoint[Y] );
-    output[Z] = kalman_process(&kalmanFilterStateRate[Z], input[Z], setPoint[Z] );
-
-    DEBUG_SET(DEBUG_KALMAN, 1, input[X]);                               //Gyro input
+    update_kalman_covariance(input, axis);
+    input = kalman_process(&kalmanFilterStateRate[axis], input, getSetpointRate(axis) );
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Assignment of function parameter has no effect outside the function.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=5382440)